conda_smithy/lint_recipe.py

```diff
@@ -515,6 +515,18 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
                 " form. See lines %s." % (bad_lines,)
             )
 
+    # 25: requirements should be all in lower-case
+    host_reqs = requirements_section.get("host") or []
+    run_reqs = requirements_section.get("run") or []
+    for reqs in [host_reqs, run_reqs]:
```

Comment:
what about build requirements and test requirements?

Reply:
Also do you have any interest in writing a mini migrator for the bot to fix this on existing packages?

Reply:
@CJ-Wright , thank you for reminding me about the build and test requirements. I added those, including the output requirement as well. 
However, there might be other combinations such as `build -> run_exports -> package`, `build -> run_exports -> strong -> package`, `build -> ignore_run_exports -> package`, `requirements -> run_constrained -> package`, `outputs -> requirements -> build -> package`, `outputs -> run_exports -> package` that have a package name listed. These are the ones I thought of, but there might be few more, and I don't know if it is worth considering including them all now.

Reply:
And regarding the mini migrator for the bot - I don't know if I am knowledgeable enough to do that, but I am definitely willing to try it if you can give me some directions to use as an example.

Reply:
here are two examples of mini-migrators:
https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/use_pip.py
https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/license.py
that you could use as a guide.

Reply:
(look for the class that inherits from MiniMigrator)